test/Test.ts

```diff
@@ -14,222 +13,336 @@ let sarcoDao: SignerWithAddress;
 let voter1: SignerWithAddress;
 let voter2: SignerWithAddress;
 
-describe("Contract: Collection", function () {
-    let initialContractBalance: BigNumber = ethers.utils.parseEther('100');
-
-    beforeEach(async () => {
-      const Sarco = await ethers.getContractFactory("Sarco");
-      const Collection = await ethers.getContractFactory("Collection");
-      // initialising signers, with 2 voters
-      [sarcoDao, tokenOwner, voter1, voter2] = await ethers.getSigners()
-      // deplying contracts
-      sarco = await Sarco.connect(tokenOwner).deploy(tokenOwner.address) as Sarco; 
-      collection = await Collection.connect(sarcoDao).deploy(sarco.address) as Collection;
-
-      // funding the Collection contract with 100 SARCO
-      await sarco.connect(tokenOwner).transfer(collection.address, initialContractBalance)
+describe('Contract: Collection', function () {
+  let initialContractBalance: BigNumber = ethers.utils.parseEther('100');
+
+  beforeEach(async () => {
+    const SarcoFactory = await ethers.getContractFactory('Sarco');
+    const CollectionFactory = await ethers.getContractFactory('Collection');
+    // initialising signers, with 2 voters
+    [sarcoDao, tokenOwner, voter1, voter2] = await ethers.getSigners();
+    // deplying contracts
+    sarco = (await SarcoFactory.connect(tokenOwner).deploy(tokenOwner.address)) as Sarco;
+    collection = (await CollectionFactory.connect(sarcoDao).deploy(sarco.address)) as Collection;
+
+    // funding the Collection contract with 100 SARCO
+    await sarco.connect(tokenOwner).transfer(collection.address, initialContractBalance);
+  });
+
+  describe('allocateRewards()', () => {
+    context('Successfully allocates rewards to voters', () => {
+      let voterReward = ethers.utils.parseEther('10');
+
+      it('should allocate rewards among 2 voters', async () => {
+        let rewards: Reward[] = [
```

Comment:
Not picky about this because I think it's acceptable how you have it but I just wanted to present this as an option. Since rewards is defined the same way in each test it could be added to a `beforeEach` function at the beginning of this context. For instance, on line 35 
```
let rewards: Reward[] = [];
beforeEach(() => {
  rewards = [
    { _address: voter1.address, _amount: voterReward },
    { _address: voter2.address, _amount: voterReward },
  ];
})

Reply:
I've tried thinking about it before. I implemented this but it's giving me:
Cannot redeclare block-scoped variable 'rewards'